### ISSUE TITLE
[FR]: Enable No-Persistence / No-Cache Queries

### ISSUE THREAD

**Issue by johnfrancmartin** [firebase/firebase-ios-sdk#12397]
### Description

**Use Case**

I'd like to be able to make certain queries explicitly with no persistence. Given large cached collections in persistence can impact performance, there are some larger documents that I'd like to query when needed but never persist.

**How the Firebase SDK currently handles it**

Persistence settings are set globally at Firestore initialization, there is currently no API to enable certain queries to sidestep cache.

**How you propose the Firebase SDK should handle it**

Either: allow the creation of alternate Firestore instances, with different persistence settings, or, allow queries to provide a `disableCaching` option, that skips persistence.

### Firebase Product(s)

Firestore

**Comment by MarkDuckworth** [comment]
Hi @johnfrancmartin, Thanks for the feature request. I'll review this use case with the team.

Can you create multiple Firestore instances via multiple Firebase app instances using [`firestore(app:)`](https://firebase.google.com/docs/reference/swift/firebasefirestore/api/reference/Classes/Firestore#firestoreapp:), and configure persistence settings for each?

**Comment by MarkDuckworth** [comment]
I reviewed this with the team and this is something we have been considering, so your request helps us prioritize it. For now, the only additional workaround I have is to get the documents you do not want to cache within a [transaction](https://firebase.google.com/docs/reference/swift/firebasefirestore/api/reference/Classes/Transaction). You will not be able to run a query within a transaction, but you can get individual documents by ID.

**Comment by johnfrancmartin** [comment]
@MarkDuckworth any updates on this?

**Comment by MarkDuckworth** [comment]
@johnfrancmartin, no, I don't have a timeline for this feature yet. We're you able to implement a workaround with a second FirebaseApp instance or using transactions?
